nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/main/java/org/nuxeo/ecm/platform/comment/impl/CommentManagerImpl.java

```diff
@@ -500,4 +460,52 @@ public boolean hasFeature(Feature feature) {
             throw new UnsupportedOperationException(feature.name());
         }
     }
+
+    @Override
+    public List<DocumentModel> getComments(CoreSession session, DocumentModel docModel) {
+        Function<CoreSession, List<DocumentModel>> getCommentsFunction = userSession -> {
+            Map<String, Object> ctxMap = Collections.<String, Object> singletonMap(
+                    ResourceAdapter.CORE_SESSION_CONTEXT_KEY, userSession);
+            RelationManager relationManager = Framework.getService(RelationManager.class);
+            Graph graph = relationManager.getGraph(config.graphName, userSession);
+            Resource docResource = relationManager.getResource(config.documentNamespace, docModel, ctxMap);
+            if (docResource == null) {
+                throw new NuxeoException("Could not adapt document model to relation resource ; "
+                        + "check the service relation adapters configuration");
+            }
+
+            // FIXME AT: why no filter on the predicate?
+            List<Statement> statementList = graph.getStatements(null, null, docResource);
+            if (graph instanceof JenaGraph) {
+                // XXX AT: BBB for when repository name was not included in the
+                // resource uri
+                Resource oldDocResource = new QNameResourceImpl(config.documentNamespace, docModel.getId());
+                statementList.addAll(graph.getStatements(null, null, oldDocResource));
+            }
+
+            List<DocumentModel> commentList = new ArrayList<>();
+            for (Statement stmt : statementList) {
+                QNameResourceImpl subject = (QNameResourceImpl) stmt.getSubject();
+
+                DocumentModel commentDocModel = (DocumentModel) relationManager.getResourceRepresentation(
+                        config.commentNamespace, subject, ctxMap);
+                if (commentDocModel == null) {
+                    // XXX AT: maybe user cannot see the comment
+                    log.warn("Could not adapt comment relation subject to a document "
+                            + "model; check the service relation adapters configur  ation");
+                    continue;
+                }
+                commentList.add(commentDocModel);
+            }
+
+            CommentSorter sorter = new CommentSorter(true);
+            Collections.sort(commentList, sorter);
+
+            return commentList;
+        };
+        if (session != null) {
+            return CoreInstance.doPrivileged(session, getCommentsFunction);
+        }
+        return CoreInstance.doPrivileged(docModel.getRepositoryName(), getCommentsFunction);
+    }
```

Comment:
to avoid the control on the `session` to chose which override `doPrivileged` we will call. I think this code can be replaced by 
```
CoreSession coreSession = session != null ? session : docModel.getCoreSession();
        return CoreInstance.doPrivileged(coreSession, getCommentsFunction);
```

Reply:
as discuss with @troger and @NourNuxeo  we will use directly:

```
return CoreInstance.doPrivileged(docModel.getRepositoryName(), getCommentsFunction);
```
without the variable `getCommentsFunction` use inline lambda